pkg/networkservice/connectioncontext/dnscontext/server.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package dnscontext provides a dns context specific chain element. It just adds dns configs into dns context of connection context.
+// It also provides a possibility to use custom dns configs getters for setup users endpoints.
+package dnscontext
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+)
+
+// GetDNSConfigsFunc gets dns configs
+type GetDNSConfigsFunc func(conn *networkservice.Connection) []*networkservice.DNSConfig
```

Comment:
Do we anticipate DNSConfig changing dynamically through the life of the server?

Reply:
I'm not sure that it should be static.
I think it can be useful if the user's DNS configs could be provided from some dynamic thing.
Can you please share your thoughts?

Reply:
I’m not clear when it would be dynamic like this from a Connection... I could see a func() []*networkservice.DNSConfig ... I don’t quite get the example from utils/test ... what am I missing?

Reply:
I thought that it could be useful for users, but actually I can imagine only one scenario of usage `ConfigFromConnection` it is just a case when the DNS server located as a sidecar in NSE POD. 
It could be used only for testing. 
Finally, I've removed it.